scvi/inference/trainer.py

```diff
@@ -171,6 +174,22 @@ def train(self, n_epochs=20, lr=1e-3, eps=0.01, params=None):
                 % (int(self.training_time), self.n_epochs)
             )
 
+    def check_training_status(self, loss):
+        loss_is_nan = torch.isnan(loss).item()
+        if loss_is_nan:
+            logger.warning("Model loss was NaN")
+            self.nan_counter += 1
+            self.previous_loss_was_nan = True
+        else:
+            self.nan_counter = 0
+            self.previous_loss_was_nan = False
+
+        if self.nan_counter >= 10:
```

Comment:
I'm not sure why, but this should probably be the choice of user (this threshold, probably in the train method?). With a default of 10 is fine.